pkg/osvscanner/osvscanner.go

```diff
@@ -888,13 +888,14 @@ func patchPackageForRequest(pkg scannedPackage) scannedPackage {
 
 func makeRequest(
 	r reporter.Reporter,
-	packages []scannedPackage,
+	packages *[]scannedPackage,
 	compareLocally bool,
 	compareOffline bool,
 	localDBPath string) (*osv.HydratedBatchedResponse, error) {
 	// Make OSV queries from the packages.
 	var query osv.BatchedQuery
-	for _, p := range packages {
+	for i := range *packages {
+		p := &(*packages)[i]
 		p = patchPackageForRequest(p)
```

Comment:
```suggestion
	for i, p := range packages {
		packages[i] = patchPackageForRequest(p)
```

Something like this should work.